plugin-cerberus/src/main/java/ca/on/oicr/gsi/shesmu/cerberus/BaseCerberusFileProvenanceSkippedRecord.java

```diff
@@ -0,0 +1,149 @@
+package ca.on.oicr.gsi.shesmu.cerberus;
+
+import ca.on.oicr.gsi.cerberus.fileprovenance.ProvenanceRecord;
+import ca.on.oicr.gsi.provenance.model.LimsProvenance;
+import ca.on.oicr.gsi.shesmu.gsicommon.CerberusFileProvenanceSkippedValue;
+import ca.on.oicr.gsi.shesmu.gsicommon.IUSUtils;
+import ca.on.oicr.gsi.shesmu.plugin.AlgebraicValue;
+import ca.on.oicr.gsi.shesmu.plugin.Tuple;
+import ca.on.oicr.gsi.vidarr.api.ProvenanceWorkflowRun;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+abstract class BaseCerberusFileProvenanceSkippedRecord<T extends LimsProvenance>
+    implements CerberusFileProvenanceSkippedValue {
+
+  static Map<String, JsonNode> labelsToMap(ProvenanceWorkflowRun<?> workflow) {
+    final var workflowRunLabels = new TreeMap<String, JsonNode>();
+    final var labels = workflow.getLabels().fields();
+    while (labels.hasNext()) {
+      final var label = labels.next();
+      workflowRunLabels.put(label.getKey(), label.getValue());
+    }
+    return workflowRunLabels;
+  }
+
+  // Do we care about stale records if they've been failed?
```

Comment:
My guess is no, but I hope @mlaszloffy can speak to this

Reply:
I think `stale` should be included so that the input format is drop in compatible with cerberus_fp and so that the records we're trying to fetch from cerberus_fp_skipped can be inspected to see if they have gone stale or not.